std-bits/table/src/main/java/org/enso/table/excel/ExcelRow.java

```diff
@@ -40,14 +39,20 @@ public Object getCellValue(int column) {
         double dblValue = cell.getNumericCellValue();
         if (DateUtil.isCellDateFormatted(cell) && DateUtil.isValidExcelDate(dblValue)) {
           var dateTime = DateUtil.getLocalDateTime(dblValue);
-          if (dateTime.isBefore(LocalDateTime.of(1900, 1, 2, 0, 0))) {
-            // Excel stores times as if they are on the 1st January 1900.
-            // Due to the 1900 leap year bug might be 31st December 1899.
-            return dateTime.toLocalTime();
-          }
-          if (dateTime.getHour() == 0 && dateTime.getMinute() == 0 && dateTime.getSecond() == 0) {
+          var dateFormat = cell.getCellStyle().getDataFormatString();
+          var hasTime =
+              dateFormat.contains("h") || dateFormat.contains("H") || dateTime.getHour() != 0;
```

Comment:
Just note that this can still cut dates like `00:12:34` because it ignores minutes and seconds. I think the last condition should be:
```suggestion
          var hasTime =
              dateFormat.contains("h") || dateFormat.contains("H") || dateTime.getHour() != 0 || dateTime.getMinute() != 0 || dateTime.getSecond() != 0;
```

Reply:
Yep good point. Fixed